src/include/duckdb/planner/expression_binder/column_alias_binder.hpp

```diff
@@ -16,18 +16,40 @@ namespace duckdb {
 class BoundSelectNode;
 class ColumnRefExpression;
 
-//! A helper binder for WhereBinder and HavingBinder which support alias as a columnref.
-class ColumnAliasBinder {
+//! A helper to allow binders to look up the index of a column alias.
+class ColumnAliasLookup {
 public:
-	ColumnAliasBinder(BoundSelectNode &node, const case_insensitive_map_t<idx_t> &alias_map);
+	ColumnAliasLookup(const case_insensitive_map_t<idx_t> &alias_map);
 
-	BindResult BindAlias(ExpressionBinder &enclosing_binder, ColumnRefExpression &expr, idx_t depth,
-	                     bool root_expression);
+	idx_t TryBindAlias(const ParsedExpression &expr);
 
-private:
-	BoundSelectNode &node;
+protected:
 	const case_insensitive_map_t<idx_t> &alias_map;
-	bool in_alias;
+};
+
+//! A helper to allow binders to resolve a column alias by duplicating the statement the alias refers to.
+class ColumnAliasBinder {
+public:
+	ColumnAliasBinder(const BoundSelectNode &node);
```

Comment:
Could you perhaps expand on why the `ColumnAliasProjectionBinder `, `ColumnAliasBinder` and `ColumnAliasLookup` are separate classes? They seem to always be used together. Or perhaps `ColumnAliasBinder` and `ColumnAliasProjectionBinder` could both inherit from `ColumnAliasLookup`?